src/pylint_odoo/checkers/import_checker.py

```diff
@@ -0,0 +1,78 @@
+from functools import lru_cache
+from pathlib import Path
+from typing import Set
+
+from astroid.nodes import ImportFrom, Module
+from pylint.checkers.utils import only_required_for_messages
+from pylint.lint import PyLinter
+
+from .odoo_base_checker import OdooBaseChecker
+
+ODOO_MSGS = {
+    # C->convention R->refactor W->warning E->error F->fatal
+    "E8401": (
+        "This python file is not imported and has no effect",
+        "missing-python-import",
+        "Import it on the corresponding __init__.py file",
+    ),
+}
+
+
+class ImportChecker(OdooBaseChecker):
+    msgs = ODOO_MSGS
+    name = "odoolint"
+
+    def __init__(self, linter: PyLinter):
+        self.imports = {}
+        self.orm_files = {}
+
+        super().__init__(linter)
+
+    @lru_cache()
+    def _get_odoo_import_orm_names(self, module: Module) -> Set[str]:
+        results = set()
+        for import_stmt in filter(lambda node: isinstance(node, ImportFrom), module.body):
+            if import_stmt.modname == "odoo":
+                if any("models" in name for name in import_stmt.names):
+                    results.add("models")
+            elif import_stmt.modname == "odoo.models":
+                results.update([name[0] for name in import_stmt.names])
+
+        return results
+
+    @only_required_for_messages("missing-python-import")
+    def visit_classdef(self, node) -> None:
+        odoo_imports = self._get_odoo_import_orm_names(node.parent)
+        if "models" in odoo_imports:
+            not_orm = not any(
+                (basename.startswith("models.") or basename in odoo_imports) for basename in node.basenames
```

Comment:
Sorry, what would be the generic check exactly? AFAIK outside of Odoo it is normal to have empty __init_.py files, so I don't think it would be helpful.

Reply:
Also I did the models thing to extend it and make sure that both controllers and models are imported in their __init__.py so they make an effect. 

If I add a utility class inside the models folder, it is not required to import it in the __init__.py, I think that is why I checked for class defs.

Reply:
I think I understand now, do you mean checking that all .py files are imported somewhere?